doc/BMI_MODELS.md

```diff
@@ -279,7 +279,7 @@ An example implementation for an appropriate BMI model as a **C++** shared libra
 
 ### Enabling Python Integration
 
-Python integration is controlled with the CMake build flag `NGEN_ACTIVATE_PYTHON`, however this currently defaults to "On"--you would need to turn this off if Python is not available in your environment. See [the Dependencies documentation](DEPENDENCIES.md#python-3-libraries) for specifics on Python requirements, but in summary you will need a working Python environment with NumPy installed. You can set up a Python environment anywhere with the usual environment variables, but note that ngen will look for a Python environment in the build directory named `.venv` or `venv` by default. The appropriate Python environment should be active in the shell when ngen is run.
+Python integration is controlled with the CMake build flag `NGEN_ACTIVATE_PYTHON`, however this currently defaults to "On"--you would need to turn this off if Python is not available in your environment. See [the Dependencies documentation](DEPENDENCIES.md#python-3-libraries) for specifics on Python requirements, but in summary you will need a working Python environment with NumPy installed. You can set up a Python environment anywhere with the usual environment variables. The appropriate Python environment should be active in the shell when ngen is run.
```

Comment:
So, should we pivot this PR just a little and also get rid of this? : https://github.com/NOAA-OWP/ngen/blob/ccce9dc1f2915b90719bff4f0f9b061367220cd2/include/utilities/python/InterpreterUtil.hpp#L217-L234
I have been reluctant to because it could theoretically break someone's working setup. However, for everything there is a season.

Reply:
Oof, I hadn't figured there was associated code doing the same. I'll take a look, but I'm kinda inclined to be a bit more cautious about that angle.

Reply:
Yeah, I've been cautious about removing this also. But with the above documentation change the code referenced above becomes a COMPLETELY undocumented and rather odd behavior. How to reconcile?

Reply:
Also related is https://github.com/NOAA-OWP/ngen/issues/441#issuecomment-1318890579

Reply:
The 'Expected Behavior' in the issue description there is pretty strong. Let's discuss on Monday, after I've had a chance to look more deeply?

Reply:
> I am not sure whether the currently-active Python environment or the build system's Python environment would be used when running a built executable (or worse, a mix of both).

My understanding (possibly flawed) is that currently the build-time impacts of the activated venv are:
1. The Python runtime for that Python version is what gets embedded with pybind11... so the _version_ of Python for that build is fixed from there forward--running that executable with an activated venv made with a different version of Python would probably cause Bad Things™️.
2. The numpy headers that are found at that time are compiled in. So in theory running that executable with a different venv containing a different version of NumPy _might_ blow up... there are probably a lot of unpredictable caveats there.

Notably what is _not_ included above is that the path to the venv that was present at build time is nowhere compiled in... generally this is probably a good thing (debate?) and not something we'd want to change. Even if we did, besides the scenario of moving that venv aside and putting a different one in it's place--which we can't control--it's possible to upgrade a venv in-place to a newer version of Python.... so we're not really accomplishing much meaningful.

> do we need to...more robustly ensure the right environment gets loaded at runtime?

The "right" environment--if this means the venv present at compilation time--is arguably indeterminable _ex post facto_, for reasons including those described above. We _could_ (and maybe should) inspect the present venv at startup and try to use some heuristics (exchanging one set of heuristics for another now) to determine whether the presently loaded Python environment (which might _not_ be a venv) is valid for the ngen build... e.g. Python micro version different = WARN, Python minor version different = FAIL? Similar checks on NumPy?

Reply:
> or worse, a mix of both...

Also, the "mix of both" is probably the biggest problem with the _current_ code... it is at least possible (if not likely) to run with one version of Python active (e.g. the system Python) and for ngen to pick up package paths from another version (e.g. the version used to create whatever is at the path `.venv`).

Lastly, attempting to "fix" (somehow) the venv that was run at build time as the one that must be used at runtime would interfere with what was trying to be done in #441 --essentially, what the submitter wanted there was the ability to switch venvs from run to run trying different versions of a module (and combinations of dependencies of that module, I think)...which seems like a useful use case and something we should try not to prevent... though it's impossible to support without caveats (all must use same Python version, for starters).

Reply:
Yes, I am fairly confident that Python itself and numpy (that ngen uses) are compiled into ngen.  You touch on my concern, though.  If the same environment isn't used at runtime, that opens the door to discrepancies with either the required version of Python or numpy for a BMI module imported from the runtime Python environment.